src/ZODB/scripts/repozo.py

```diff
@@ -755,6 +746,88 @@ def do_recover(options):
             raise
 
 
+def do_incremental_recover(options, repofiles):
+    datfile = os.path.splitext(repofiles[0])[0] + '.dat'
+    log('Recovering (incrementally) file to %s', options.output)
+    with open(datfile) as fp, open(options.output, 'r+b') as outfp:
+        outfp.seek(0, 2)
+        initial_length = outfp.tell()
+        previous_chunk = None
+        for line in fp:
+            fn, startpos, endpos, _ = chunk = line.split()
+            startpos = int(startpos)
+            endpos = int(endpos)
+            if endpos > initial_length:
+                break
+            previous_chunk = chunk
+        else:
+            if endpos == initial_length:
+                log('Target file is same size as latest backup, '
+                    'doing nothing.')
+                return
+            else:
+                log('Target file is longer than latest backup, '
+                    'falling back to a full recover.')
+                return do_full_recover(options, repofiles)
```

Comment:
It seems the [test failure on windows](https://github.com/zopefoundation/ZODB/actions/runs/11452216534/job/31862720635#step:6:85) is because of here, we have opened `options.output` already and `do_full_recover` would re-open it a second time, which is not allowed on windows as far as I remember